### PR TITLE
Side panels materialfx

### PR DESCRIPTION
**Pull request by Anastasia-Nikitina**

### CODE REVIEW

File: src/main/kotlin/solve/main/MainView.kt [hunk 1]
```diff
@@ -47,9 +70,105 @@ class MainView : View() {
...
+        styleClass.add("mfxButton")
+
+        setPrefSize(Style.navigationRailSize, Style.navigationRailSize)
+
+        padding = Insets(0.0, 0.0,0.0,0.0)
+
+        contentDisplay = ContentDisplay.TOP
+        graphic = ImageView(helpIcon)
+
+        style = "-fx-font-family: ${Style.font}; -fx-font-weight:700;  -fx-font-size: ${Style.buttonFontSize}; -fx-text-fill: ${Style.primaryColorLight}; -fx-background-radius: 36"
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: This code seems to be common for all side panel buttons (help, settings, etc.)

File: src/main/kotlin/solve/main/MainView.kt [hunk 2]
```diff
@@ -47,9 +70,105 @@ class MainView : View() {
...
+    private val leftPanel = vbox(7){
+        addStylesheet(MFXButtonStyleSheet::class)
+        style = "-fx-background-color: #${Style.surfaceColor}"
+        add(nameApp)
+        add(importFAB)
+        add(leftSidePanelViews.tabsView.root)
+        add(pluginsButton)
+        add(settingsButton)
+        add(helpButton)
+
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: space

File: src/main/kotlin/solve/main/MainView.kt [hunk 3]
```diff
@@ -68,12 +187,19 @@ class MainView : View() {
...
 
     override val root = mainViewBorderPane
 
+    private fun importAction(){
+        controller.directoryPath.set(null)
+        controller.projectAfterPartialParsing.set(null)
+        content = MaterialFXDialog.createGenericDialog(ImporterView().root)
+        dialog = MaterialFXDialog.createStageDialog(content, find<MainView>().currentStage, find<MainView>().root)
+        dialog.show()
+        content.padding = Insets(0.0,0.0,10.0,0.0)
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: The same as for other PR

File: src/main/kotlin/solve/styles/MFXButtonStyleSheet.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+                backgroundColor += Paint.valueOf(Style.backgroundColour)
+
+            }
+            and(pressed){
+                backgroundColor += Paint.valueOf(Style.surfaceColor)
+            }
+       }
+    }
+
+
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/styles/MFXButtonStyleSheet.kt]: spaces

File: src/main/kotlin/solve/constants/ResourcesPaths.kt [hunk 5]
```diff
@@ -8,12 +8,19 @@ const val IconsImporterErrorFilePath = "icons/importer/error_file.png"
 const val IconsImporterErrorFolderPath = "icons/importer/error_folder.png"
 const val IconsImporterPhotoPath = "icons/importer/photo.png"
 const val IconsImporterWarningPath = "icons/importer/warning.png"
-const val IconsSidePanelCataloguePath = "icons/sidepanel/catalogue.png"
-const val IconsSidePanelVisualizationSettingsPath = "icons/sidepanel/visualization_settings.png"
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/constants/ResourcesPaths.kt:12]: Did you remove these icons?

File: src/main/kotlin/solve/importer/view/ControlPanel.kt [hunk 6]
```diff
@@ -153,7 +150,7 @@ class ControlPanel : View() {
     }
 
     private fun showLoading() {
-        MaterialFXDialog.changeContent(find<MenuBarView>().content, LoadingScreen().root)
+        MaterialFXDialog.changeContent(find<MainView>().content, LoadingScreen().root)
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/importer/view/ControlPanel.kt]: Move MainView into the field and get it by inject()

Comment (Alexander-Ploskin) [reply src/main/kotlin/solve/importer/view/ControlPanel.kt]: The same for LoadingScreen

File: src/main/kotlin/solve/importer/view/ControlPanel.kt [hunk 7]
```diff
@@ -162,8 +159,7 @@ class ControlPanel : View() {
             mainController.displayCatalogueFrames(projectVal.frames)
             button.isDisable = true
 
-            find<MenuBarView>().dialog.close()
-
+            find<MainView>().dialog.close()
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/importer/view/ControlPanel.kt]: Move MainView into the field and get it by inject()

File: src/main/kotlin/solve/importer/view/LoadingScreen.kt [hunk 8]
```diff
@@ -31,36 +31,39 @@ class LoadingScreen : View("Loading") {
...
-                        close()
+                borderpane {
+                    right {
+                        mfxButton("CANCEL") {
+                            BorderPane.setMargin(this, Insets(0.0, 24.0, 24.0, 0.0))
+                            maxWidth = 75.0
+                            prefHeight = 23.0
+                            style = Style.buttonStyle
+                            action {
+                                find<ControlPanel>().coroutineScope.cancel()
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/importer/view/LoadingScreen.kt]: Move ControlPanel into the field and get it by inject()

File: src/main/kotlin/solve/main/MainView.kt [hunk 9]
```diff
@@ -25,20 +36,32 @@ class MainView : View() {
...
         private const val TabsViewInitialTabParamName = "initialTab"
+
+        private val importIcon = loadResourcesImage(IconsImportFAB)
+        private val pluginsIcon = loadResourcesImage(IconsPlugins)
+        private val settingsIcon = loadResourcesImage(IconsSettings)
+        private val helpIcon = loadResourcesImage(IconsHelp)
+        private val gridIcon = loadResourcesImage(IconsGrid)
     }
 
+    val importer = find<ImporterView>()
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: by inject

File: src/main/kotlin/solve/main/MainView.kt [hunk 10]
```diff
@@ -47,9 +70,105 @@ class MainView : View() {
     private val rightSidePanelViews =
         createSidePanelsViews(rightSidePanelTabs, SidePanelLocation.Right)
 
+    private val importFAB = mfxButton {
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: naming

File: src/main/kotlin/solve/main/MainView.kt [hunk 11]
```diff
@@ -47,9 +70,105 @@ class MainView : View() {
...
+        VBox.setMargin(this, Insets(5.0, 8.0,10.0,8.0))
+
+        val circle = Circle(this.layoutX + 28.0, this.layoutY + 28.0,28.0)
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: space, move radius into a const

File: src/main/kotlin/solve/main/MainView.kt [hunk 12]
```diff
@@ -47,9 +70,105 @@ class MainView : View() {
     private val rightSidePanelViews =
         createSidePanelsViews(rightSidePanelTabs, SidePanelLocation.Right)
 
+    private val importFAB = mfxButton {
+        VBox.setMargin(this, Insets(5.0, 8.0,10.0,8.0))
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: spaces

File: src/main/kotlin/solve/main/MainView.kt [hunk 13]
```diff
@@ -47,9 +70,105 @@ class MainView : View() {
...
+        prefWidth = 56.0
+
+        style = "-fx-background-color: #${Style.secondaryColor}; -fx-background-radius: 28;"
+
+        action{
+            importAction()
+        }
+    }
+
+    private val pluginsButton = mfxButton("Plugins") {
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: Don't show unused tabs (plugins, settings, help) before the better times (don't remove the code and icons)

File: src/main/kotlin/solve/main/MainView.kt [hunk 14]
```diff
@@ -47,9 +70,105 @@ class MainView : View() {
...
+        clip = Style.circleForRipple(this)
+        contentDisplay = ContentDisplay.TOP
+        setPrefSize(Style.navigationRailSize, Style.navigationRailSize)
+        styleClass.add("mfxButton")
+        prefHeight = 70.0
+        prefWidth = 70.0
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: Why you set size only for this button? It actually affects background circle
image
image

File: src/main/kotlin/solve/styles/MFXButtonStyleSheet.kt [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+    init{
+        mfxButton{
+            and(hover, pressed){
+                backgroundColor += Paint.valueOf(Style.backgroundColour)
+
+            }
+            and(pressed){
+                backgroundColor += Paint.valueOf(Style.surfaceColor)
+            }
+       }
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/styles/MFXButtonStyleSheet.kt]: codestyle

File: src/main/kotlin/solve/main/MainView.kt [hunk 16]
```diff
@@ -47,9 +70,105 @@ class MainView : View() {
...
+        padding = Insets(0.0, 0.0,0.0,0.0)
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: can use paddingAll

File: src/main/kotlin/solve/styles/Style.kt [hunk 17]
```diff
@@ -26,4 +31,8 @@ object Style {
         const val headerFontSize = "20px"
 
         const val tooltipFontSize = "12px"
+
+        const val navigationRailSize = 72.0
+
+        fun circleForRipple(button: MFXButton) = Circle(button.layoutX + 36.0, button.layoutY + 36.0,35.0)
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/styles/Style.kt]: space

**Review by Alexander-Ploskin — CHANGES_REQUESTED** [review]

File: src/main/kotlin/solve/main/MainView.kt [hunk 18]
```diff
@@ -47,9 +70,105 @@ class MainView : View() {
...
+
+        contentDisplay = ContentDisplay.TOP
+        graphic = ImageView(helpIcon)
+
+        style = "-fx-font-family: ${Style.font}; -fx-font-weight:700;  -fx-font-size: ${Style.buttonFontSize}; -fx-text-fill: ${Style.primaryColorLight}; -fx-background-radius: 36"
+    }
+
+    private val nameApp = label("SOLVE") {
+        style = "-fx-font-family: ${Style.font}; -fx-font-weight:700; -fx-font-size: 18px"
+        VBox.setMargin(this, Insets(12.0, 6.0, 0.0, 6.0))
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: Wrong top margin

Desired:
image

Actual:
image

**Review by Alexander-Ploskin — CHANGES_REQUESTED** [review]

File: src/main/kotlin/solve/main/MainView.kt (hunk 1 above)
Comment (Anastasia-Nikitina) [reply src/main/kotlin/solve/main/MainView.kt]: this is fixed

File: src/main/kotlin/solve/main/MainView.kt (hunk 3 above)
Comment (Anastasia-Nikitina) [reply src/main/kotlin/solve/main/MainView.kt]: this is fixed

File: src/main/kotlin/solve/constants/ResourcesPaths.kt (hunk 5 above)
Comment (Anastasia-Nikitina) [reply src/main/kotlin/solve/constants/ResourcesPaths.kt:12]: this is fixed

File: src/main/kotlin/solve/main/MainView.kt [hunk 19]
```diff
@@ -68,12 +162,19 @@ class MainView : View() {
...
-        left = leftSidePanelViews.tabsView.root
-        right = rightSidePanelViews.tabsView.root
     }
 
     override val root = mainViewBorderPane
 
+    private fun importAction() {
+        controller.directoryPath.set(null)
+        controller.projectAfterPartialParsing.set(null)
+        content = MaterialFXDialog.createGenericDialog(ImporterView().root)
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/main/MainView.kt]: by inject()

**Review by Alexander-Ploskin — CHANGES_REQUESTED** [review]

File: src/main/kotlin/solve/importer/ProjectParser.kt [hunk 20]
```diff
@@ -152,6 +152,7 @@ object ProjectParser {
         if (!isImagesExist) {
             createAlertForError("The images folder is missing in the directory")
             return null
+
```
Comment (Alexander-Ploskin) [inline src/main/kotlin/solve/importer/ProjectParser.kt]: space

**Review by Alexander-Ploskin — APPROVED** [review]

**Review by DmiitriyJarosh — APPROVED** [review]

**Pull request merged**